components/director/internal/domain/application/repository.go

```diff
@@ -213,3 +240,17 @@ func (r *pgRepository) multipleFromEntities(entities EntityCollection) ([]*model
 	}
 	return items, nil
 }
+
+func updateOperationContext(ctx context.Context, appID string) error {
+	operations, exists := operation.FromCtx(ctx)
+	if !exists {
+		return apperrors.NewInternalError("unable to fetch operations from context")
+	}
+
+	op := (*operations)[len(*operations)-1]
+
+	op.ResourceID = appID
```

Comment:
lets have that on a central place instead - otherwise we have to add these lines to each repo. Proposal is to put this in the directive (see the other comment on the directive). Added benefit (possible as we currently wont be needing related resoureces) is that we will have the full operation construction happening in the directive